src/main/java/org/opensearch/flowframework/util/ApiSpecFetcher.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.flowframework.exception.ApiSpecParseException;
+import org.opensearch.rest.RestRequest;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.List;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.media.Content;
+import io.swagger.v3.oas.models.media.MediaType;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+import io.swagger.v3.parser.core.models.ParseOptions;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+
+/**
+ * Utility class for fetching and parsing OpenAPI specifications.
+ */
+public class ApiSpecFetcher {
+    private static final Logger LOGGER = LogManager.getLogger(ApiSpecFetcher.class);
+    private static final ParseOptions PARSE_OPTIONS = new ParseOptions();
+    private static final OpenAPIV3Parser PARSER = new OpenAPIV3Parser();
+
+    /**
+     * Default constructor for ApiSpecFetcher.
+     * It sets up default parse options for resolving references.
+     */
+    public ApiSpecFetcher() {
+        PARSE_OPTIONS.setResolve(true);
+        PARSE_OPTIONS.setResolveFully(true);
+    }
+
+    /**
+     * Parses the OpenAPI specification directly from the URI.
+     *
+     * @param apiSpecUri URI to the API specification (can be file path or web URI).
+     * @return Parsed OpenAPI object.
+     * @throws ApiSpecParseException If parsing fails.
+     */
+    public OpenAPI fetchApiSpec(URI apiSpecUri) {
```

Comment:
Looking at the test classes, we always create this URI from a String, then you spend a lot of time undoing the string in this method.  Do we need a URI argument here or can we just take a string?

Reply:
I think string would be good, I simply followed the pattern in opensearch-java before.

Reply:
+1 for copying existing code, but they probably have other external API requirements/integrations expecting that parameter.  I don't mind leaving it as-is in the interest of keeping code consistent (although a comment pointing to the source of the code for later cross-referencing would be nice).